Prevent non-admins removing admin-role

ThomasRoehl opened the pull request:
closes issue #293 

We decided to ignore the `assign-admin-role` permission and only allow removing or adding the `admin-role` for `admin-users`

For some reasons the user-edit dialog does not use the meteor methods or functions from the `RocketChat.authorization` package instead has its own method to handle that case separately. Therefore it might be possible to change the `admin-role` without being admin but with `assign-admin-role` permission from other sections of the code (e.g. via api).

mrsimpson commented on packages/rocketchat-lib/server/functions/saveUser.js:
```diff
@@ -27,7 +34,7 @@ RocketChat.saveUser = function(userId, userData) {
 	}
 
-	if (userData.roles && _.indexOf(userData.roles, 'admin') >= 0 && !RocketChat.authz.hasPermission(userId, 'assign-admin-role')) {
+	if (userData.roles && _.indexOf(userData.roles, 'admin') >= 0 && !RocketChat.authz.hasRole('admin')) {
```
Not validating the permission but the role dis-allows a non-admin with this permission to assign admins.
I agree that this permission is kind-of futile, but still I don't see a need to change this.

ThomasRoehl commented:
I removed the `assign_admin_role` permission completely.
I split it up into 2 separated commits, the first for the technical usage of the permission and the second for removing all the i18n entries of this permission. 
If we don't need to remove the i18n entries or do this differently, we can just reject the commit.
@mrsimpson please decide how to proceed.

ruKurz commented:
https://github.com/RocketChat/Rocket.Chat/issues/10494

@ThomasRoehl could you please open a PR to Rocket.Chat
